src/test/java/org/folio/services/CallNumberUtilsTest.java

```diff
@@ -48,20 +48,67 @@ public void inputForShelvingNumber(
+    "PN 12 A6 41999,PN12 .A6 41999,,PN2 .A6 1999,,,,,",
+    "PN 12 A6 41999 CD,PN12 .A6 41999 CD,,PN2 .A6 1999,,,,,CD",
+    "PN 12 A6 41999 12,PN12 .A6 41999 C.12,,PN2 .A6 1999,,,,2,",
+    "PN 12 A69 41922 12,PN12 .A69 41922 C.12,,PN2 .A69,,,1922,2,",
+    "PN 12 A69 NO 12,PN12 .A69 NO.12,,PN2 .A69,,no. 2,,,",
+    "PN 12 A69 NO 12 41922 11,PN12 .A69 NO.12 41922 C.11,,PN2 .A69,,no. 2,1922,1,",
+    "PN 12 A69 NO 12 41922 12,PN12 .A69 NO.12 41922 C.12,Wordsworth,PN2 .A69,,no. 2,1922,2,",
+    "PN 12 A69 V 11 NO 11,PN12 .A69 V.11 NO.11,,PN2 .A69,v.1,no. 1,,,",
+    "PN 12 A69 V 11 NO 11 +,PN12 .A69 V.11 NO.11 +,Over,PN2 .A69,v.1,no. 1,,,+",
+    "PN 12 A69 V 11 NO 11 41921,PN12 .A69 V.11 NO.11 41921,,PN2 .A69,v.1,no. 1,1921,,",
+    "PR 49199.3 41920 L33 41475 A6,PR 49199.3 41920 .L33 41475 .A6,,PR9199.3 1920 .L33 1475 .A6,,,,,",
+    "PQ 42678 K26 P54,PQ 42678 .K26 P54,,PQ2678.K26 P54,,,,,",
+    "PQ 48550.21 R57 V5 41992,PQ 48550.21 .R57 V15 41992,,PQ8550.21.R57 V5 1992,,,,,",
+    "PQ 48550.21 R57 V5 41992,PQ 48550.21 .R57 V15 41992,,PQ8550.21.R57 V5,,,1992,,",
+    "PR 3919 L33 41990,PR 3919 .L33 41990,,PR919 .L33 1990,,,,,",
+    "PR 49199 A39,PR 49199 .A39,,PR9199 .A39,,,,,",
+    "PR 49199.48 B3,PR 49199.48 .B3,,PR9199.48 .B3,,,,,"
+  })
+  @Test
+  public void shouldReturnsItemsWithShelvingNumber(
+    String desiredShelvingOrder,
+    String initiallyDesiredShelvesOrder,
+    String prefix,
+    String callNumber,
+    String volume,
+    String enumeration,
+    String chronology,
+    String copy,
+    String suffix
+  ) {
+    Item item = prepareAnItem(prefix, callNumber, volume, enumeration, chronology, copy, suffix);
+    HoldingsRecord holdingsRecord = new HoldingsRecord();
+    EffectiveCallNumberComponentsUtil.setCallNumberComponents(item, holdingsRecord);
 
+    assertThat(EffectiveCallNumberComponentsUtil.getCalculateAndSetEffectiveShelvingOrder(item)
+      .getEffectiveShelvingOrder(), is(desiredShelvingOrder));
+  }
+  
+  private Item prepareAnItem(String callNumberPrefix, String callNumber, String volume,
+      String enumeration, String chronology, String copyNumber, String callNumberSuffix) {
     Item item = new Item();
-    item.setItemLevelCallNumberPrefix(prefix);
+    item.setItemLevelCallNumberPrefix(callNumberPrefix);
     item.setItemLevelCallNumber(callNumber);
     item.setVolume(volume);
     item.setEnumeration(enumeration);
     item.setChronology(chronology);
-    item.setCopyNumber(copy);
-    item.setItemLevelCallNumberSuffix(suffix);
+    item.setCopyNumber(copyNumber);
+    item.setItemLevelCallNumberSuffix(callNumberSuffix);
 
-    HoldingsRecord holdingsRecord = new HoldingsRecord();
-    EffectiveCallNumberComponentsUtil.setCallNumberComponents(item,holdingsRecord);
-    EffectiveCallNumberComponentsUtil.calculateAndSetEffectiveShelvingOrder(item);
-
-    assertThat(item.getEffectiveShelvingOrder(),is (desiredShelvingOrder));
+    return item;
   }
+  
 }
```

Comment:
More tests needed. Add a test where you call tenant init or loadData and check that if your code fails, then in turn loadData fails or succeeds as you'd expect.

Reply:
It looks it's need to try to test directly methods from added shelving order update class, because relying on the flow leveraging tenant API looks quite unstable and time consuming from point of view test execution.